src/libsyncengine/jobs/network/upload_session/uploadsessiondrive.cpp

```diff
@@ -0,0 +1,120 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "uploadsessiondrive.h"
+#include "utility/utility.h"
+
+namespace KDC {
+
+UploadSessionDrive::UploadSessionDrive(int driveDbId, std::shared_ptr<SyncDb> syncDb, const SyncPath &filepath,
+                                       const SyncName &filename, const NodeId &remoteParentDirId, SyncTime modtime,
+                                       bool liteSyncActivated, uint64_t nbParalleleThread /*= 1*/)
+    : AbstractUploadSession(filepath, filename, nbParalleleThread),
+      _remoteParentDirId(remoteParentDirId),
+      _liteSyncActivated(liteSyncActivated),
+      _modtimeIn(modtime),
+      _syncDb(syncDb),
+      _driveDbId(driveDbId) {
+    _uploadSessionType = UploadSessionType::Standard;
+}
+
+UploadSessionDrive::~UploadSessionDrive() {
+    if (_vfsForceStatus) {
+        if (!_vfsForceStatus(getFilePath(), false, 100, true)) {
+            LOGW_WARN(getLogger(), L"Error in vfsForceStatus: " << Utility::formatSyncPath(getFilePath()).c_str());
+        }
+    }
+}
+
+bool UploadSessionDrive::runJobInit() {
+    if (_vfsForceStatus) {
+        if (!_vfsForceStatus(getFilePath(), true, 0, true)) {
+            LOG_WARN(getLogger(), L"Error in vfsForceStatus: " << Utility::formatSyncPath(getFilePath()).c_str());
+        }
+    }
+    return true;
+}
+
+std::shared_ptr<UploadSessionStartJob> UploadSessionDrive::createStartJob() {
+    return std::make_shared<UploadSessionStartJob>(UploadSessionType::Standard, _driveDbId, getFileName(), getFileSize(),
+                                                   _remoteParentDirId, getTotalChunks());
+}
+
+std::shared_ptr<UploadSessionChunkJob> UploadSessionDrive::createChunkJob(const std::string &chunckContent, uint64_t chunkNb,
+                                                                          std::streamsize actualChunkSize) {
+    return std::make_shared<UploadSessionChunkJob>(UploadSessionType::Standard, _driveDbId, getFilePath(), getSessionToken(),
+                                                   chunckContent, chunkNb, actualChunkSize, jobId());
+}
+
+std::shared_ptr<UploadSessionFinishJob> UploadSessionDrive::createFinishJob() {
+    return std::make_shared<UploadSessionFinishJob>(UploadSessionType::Standard, _driveDbId, getFilePath(), getSessionToken(),
+                                                    getTotalChunkHash(), getTotalChunks(), _modtimeIn);
+}
+
+std::shared_ptr<UploadSessionCancelJob> UploadSessionDrive::createCancelJob() {
+    return std::make_shared<UploadSessionCancelJob>(UploadSessionType::Standard, _driveDbId, getFilePath(), getSessionToken());
+}
+
+bool UploadSessionDrive::prepareChunkJob(const std::shared_ptr<UploadSessionChunkJob> &chunkJob) {
+    if (_liteSyncActivated) {
+        // Set VFS callbacks
```

Comment:
I would move this comment into the header file.

Reply:
[8438744](https://github.com/Infomaniak/desktop-kDrive/pull/139/commits/84387443f3f66535561db9fcf3ec96be88834542)

Reply:
[8438744](https://github.com/Infomaniak/desktop-kDrive/pull/139/commits/84387443f3f66535561db9fcf3ec96be88834542)